app/cdap/components/ConfigurationGroup/PropertyRow/index.tsx

```diff
@@ -92,6 +94,8 @@ class PropertyRowView extends React.Component<IPropertyRowProps, IState> {
 
   public state = {
     isMacroTextbox: this.isMacroTextbox(),
+    prevValue: '',
+    prevMacroValue: '${}',
   };
 
   public shouldComponentUpdate(nextProps) {
```

Comment:
oh wow the github collapsed drawer made it look like you were doing this in shouldComponentUpdate haha